contrib/win32/win32compat/misc.c

```diff
@@ -249,36 +248,29 @@ w32_fopen_utf8(const char *input_path, const char *mode)
 		return NULL; 
 	}
 
-	path = resolved_path(input_path);
-
 	/* if opening null device, point to Windows equivalent */
-	if (0 == strncmp(path, NULL_DEVICE, strlen(NULL_DEVICE)+1)) {
-		if ((r = wcsncpy_s(wpath, PATH_MAX, L"NUL", 3)) != 0) {
-			debug3("wcsncpy_s failed with error: %d.", r);
-			return NULL;
-		}
-	}
-	else
-		status = MultiByteToWideChar(CP_UTF8, 0, path, -1, wpath, PATH_MAX);
+	if (strcmp(input_path, NULL_DEVICE) == 0)
+		input_path = NULL_DEVICE_WIN;
```

Comment:
The 'const' part refers to the characters themselves being constant, not the pointer.  If we wanted the pointer and string to be constant, it would be declared "const char * const input_path".

Reply:
https://stackoverflow.com/questions/4949254/const-char-const-versus-const-char

Reply:
Exactly.  That is consistent with my explanation.